modules/pubmaticBidAdapter.js

```diff
@@ -0,0 +1,295 @@
+const constants = require('src/constants.json');
+
+const BIDDER_CODE = 'pubmatic';
+const ENDPOINT = '//hbopenbid.pubmatic.com/translator?source=prebid-client';
+const USYNCURL = '//ads.pubmatic.com/AdServer/js/showad.js#PIX&kdntuid=1&p=';
+const CURRENCY = 'USD';
+const AUCTION_TYPE = 1;
+const UNDEFINED = undefined;
+const CUSTOM_PARAMS = {
+  'kadpageurl': '', // Custom page url
+  'gender': '', // User gender
+  'yob': '', // User year of birth
+  'lat': '', // User location - Latitude
+  'lon': '', // User Location - Longitude
+  'wiid': '', // OpenWrap Wrapper Impression ID
+  'profId': '', // OpenWrap Legacy: Profile ID
+  'verId': '' // OpenWrap Legacy: version ID
+};
+const NET_REVENUE = false;
+
+let publisherId = 0;
+
+function _getDomainFromURL(url) {
+  let anchor = document.createElement('a');
+  anchor.href = url;
+  return anchor.hostname;
+}
+
+function _parseSlotParam(paramName, paramValue) {
+  if (!utils.isStr(paramValue)) {
+    paramValue && utils.logWarn('PubMatic: Ignoring param key: ' + paramName + ', expects string-value, found ' + typeof paramValue);
+    return UNDEFINED;
+  }
+
+  switch (paramName) {
+    case 'pmzoneid':
+      return paramValue.split(',').slice(0, 50).map(id => id.trim()).join();
+    case 'kadfloor':
+      return parseFloat(paramValue) || UNDEFINED;
+    case 'lat':
+      return parseFloat(paramValue) || UNDEFINED;
+    case 'lon':
+      return parseFloat(paramValue) || UNDEFINED;
+    case 'yob':
+      return parseInt(paramValue) || UNDEFINED;
+    default:
+      return paramValue;
+  }
+}
+
+function _cleanSlot(slotName) {
+  if (utils.isStr(slotName)) {
+    return slotName.replace(/^\s+/g, '').replace(/\s+$/g, '');
+  }
+  return '';
+}
+
+function _parseAdSlot(bid) {
+  bid.params.adUnit = '';
+  bid.params.adUnitIndex = '0';
+  bid.params.width = 0;
+  bid.params.height = 0;
+
+  bid.params.adSlot = _cleanSlot(bid.params.adSlot);
+
+  var slot = bid.params.adSlot;
+  var splits = slot.split(':');
+
+  slot = splits[0];
+  if (splits.length == 2) {
+    bid.params.adUnitIndex = splits[1];
+  }
+  splits = slot.split('@');
+  if (splits.length != 2) {
+    utils.logWarn('AdSlot Error: adSlot not in required format');
+    return;
+  }
+  bid.params.adUnit = splits[0];
+  splits = splits[1].split('x');
+  if (splits.length != 2) {
+    utils.logWarn('AdSlot Error: adSlot not in required format');
+    return;
+  }
+  bid.params.width = parseInt(splits[0]);
+  bid.params.height = parseInt(splits[1]);
+}
+
+function _initConf() {
+  var conf = {};
+  conf.pageURL = utils.getTopWindowUrl();
+  conf.refURL = utils.getTopWindowReferrer();
+  return conf;
+}
+
+function _handleCustomParams(params, conf) {
+  // istanbul ignore else
+  if (!conf.kadpageurl) {
+    conf.kadpageurl = conf.pageURL;
+  }
+
+  var key, value, entry;
+  for (key in CUSTOM_PARAMS) {
+    // istanbul ignore else
+    if (CUSTOM_PARAMS.hasOwnProperty(key)) {
+      value = params[key];
+      // istanbul ignore else
```

Comment:
Are these types of comments still needed?  I assume they were part of the development process and wanted to confirm.